app/src/main/java/com/github/multimatum_team/multimatum/activity/GroupDetailsActivity.kt

```diff
@@ -0,0 +1,213 @@
+
+        // Get the necessary widgets
+        groupNameView = findViewById(R.id.group_details_name)
+        groupOwnerView = findViewById(R.id.group_details_owner)
+        groupMembersView = findViewById(R.id.group_details_members)
+
+        groupInviteButton = findViewById(R.id.group_details_invite_button)
+        groupDeleteOrLeaveButton = findViewById(R.id.group_details_delete_or_leave_button)
+
+        // Initialize UI widgets
+        initTextInput()
+        initGroupMemberView()
+        initLeaveButton()
+    }
+
+    /**
+     * Update view when remove data changed.
+     */
+    private fun updateView() {
+        groupNameView.focusable = if (currentUserIsGroupOwner()) {
+            View.FOCUSABLE_AUTO
+        } else {
+            View.NOT_FOCUSABLE
+        }
+
+        groupNameView.text = SpannableStringBuilder.valueOf(group.name)
+        val ownerName = runBlocking { userRepository.fetch(group.owner).name }
+        groupOwnerView.text = getString(R.string.group_owner, ownerName)
+
+        if (currentUserIsGroupOwner()) {
+            initDeleteButton()
+        } else {
+            initLeaveButton()
+        }
+    }
+
+    /**
+     * Returns whether the group we are seeing is owned by the authenticated user.
+     */
+    private fun currentUserIsGroupOwner(): Boolean =
+        group.owner == authViewModel.getUser().value!!.id
+
+    /**
+     * Setup the text input
+     */
+    private fun initTextInput() {
+        // Adding a listener to handle the "DONE" key pressed.
+        groupNameView.setOnIMEActionDone(this) { newName ->
+            groupViewModel.renameGroup(groupID, newName)
+        }
+    }
+
+    /**
+     * Initialize group member view, setting its adapter and observing the group repository when
+     * remote data changes.
+     */
+    private fun initGroupMemberView() {
+        val adapter =
+            GroupMemberAdapter(this, userRepository, authViewModel, groupViewModel)
+
+        groupMembersView.adapter = adapter
+        groupMembersView.layoutManager = LinearLayoutManager(this)
+
+        groupViewModel.getGroups().observe(this) { groups ->
+            when (groups[groupID]) {
+                null -> finish()
+                else -> {
+                    group = groups[groupID]!!
+                    adapter.setGroup(group)
+                    updateView()
+                }
+            }
+        }
+    }
+
+    /**
+     * Initialize groupDeleteOrLeaveButton with the right text and leave the group when clicked.
+     */
+    private fun initLeaveButton() {
+        groupDeleteOrLeaveButton.text = getString(R.string.group_leave)
+        groupDeleteOrLeaveButton.setOnClickListener {
+            AlertDialog.Builder(this)
+                .setMessage(getString(R.string.group_leave_confirmation_dialog))
+                .setPositiveButton(getString(R.string.group_leave_confirm),
+                    DialogInterface.OnClickListener { dialog, which ->
```

Comment:
If I remember it well, `DialogInterface.OnClickListener` is not necessary (it worked without it in `AddDeadlineActivity` or `DeadlineDetailsActivity` )